pkg/payload/payload.go

```diff
@@ -293,47 +303,42 @@ func ValidateDirectory(dir string) error {
 	return nil
 }
 
-type payloadTasks struct {
-	idir       string
-	preprocess func([]byte) ([]byte, error)
-	skipFiles  sets.Set[string]
-}
-
-func loadUpdatePayloadMetadata(dir, releaseImage, clusterProfile string) (*Update, []payloadTasks, error) {
-	klog.V(2).Infof("Loading updatepayload from %q", dir)
-	if err := ValidateDirectory(dir); err != nil {
-		return nil, nil, err
-	}
-	var (
-		cvoDir     = filepath.Join(dir, CVOManifestDir)
-		releaseDir = filepath.Join(dir, ReleaseManifestDir)
-	)
-
-	release, arch, err := loadReleaseFromMetadata(releaseDir)
+func loadPayloadMetadata(releaseDir, releaseImage string) (*Update, error) {
+	release, err := loadReleaseMetadata(releaseDir)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 	release.Image = releaseImage
 
 	imageRef, err := loadImageReferences(releaseDir)
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	if imageRef.Name != release.Version {
-		return nil, nil, fmt.Errorf("Version from %s (%s) differs from %s (%s)", imageReferencesFile, imageRef.Name, cincinnatiJSONFile, release.Version)
+	arch := string(release.Architecture)
+	if arch == "" {
+		arch = runtime.GOARCH
+		klog.V(2).Infof("Architecture from %s (%s) retrieved from runtime: %q", cincinnatiJSONFile, release.Version, arch)
```

Comment:
Moved here from `loadReleaseFromMetadata`

Reply:
nit: I would move this part right after line 307.
Then I would not need to check if `arch` is used between 307 and 318.

Reply:
that's a good point, I'll move it right after the error check in a followup PR #1188